src/annotator/range-util.js

```diff
@@ -14,49 +14,43 @@ export function isSelectionBackwards(selection) {
 }
 
 /**
- * Returns true if `node` lies within a range.
- *
- * This is a simplified version of `Range.isPointInRange()` for compatibility
- * with IE.
+ * Returns true if any part of `node` lies within `range`.
  *
  * @param {Range} range
  * @param {Node} node
  */
 export function isNodeInRange(range, node) {
-  if (node === range.startContainer || node === range.endContainer) {
-    return true;
+  try {
+    const length = node.nodeValue?.length ?? node.childNodes.length;
+    return (
+      // Check start of node is before end of range.
+      range.comparePoint(node, 0) <= 0 &&
+      // Check end of node is after start of range.
+      range.comparePoint(node, length) >= 0
+    );
+  } catch (e) {
+    // `comparePoint` may fail if the `range` and `node` do not share a common
+    // ancestor or `node` is a doctype.
+    return false;
   }
```

Comment:
[`Range.comparePoint`](https://developer.mozilla.org/en-US/docs/Web/API/Range/comparePoint) is available in Edge 17 and above, as well as all other engines. This fortunately matches our new browser support baseline :)

Reply:
Oh, you answered my question!